src/rust/cryptography-x509/src/pkcs7.rs

```diff
@@ -29,22 +29,38 @@ pub enum Content<'a> {
     EncryptedData(asn1::Explicit<EncryptedData<'a>, 0>),
 }
 
-#[derive(asn1::Asn1Write)]
+#[derive(asn1::Asn1Write, asn1::Asn1Read)]
 pub struct SignedData<'a> {
     pub version: u8,
-    pub digest_algorithms: asn1::SetOfWriter<'a, common::AlgorithmIdentifier<'a>>,
+    pub digest_algorithms: common::Asn1ReadableOrWritable<
```

Comment:
Would you be up for pulling the `Asn1ReadableOrWritable` pieces out into a separate PR? I think this is very very close, and I'm looking for ways to keep it moving as I review the last pieces.

Reply:
Don, #11922 is opened!